packages/rif-id-core/src/operations/authentication.ts

```diff
@@ -0,0 +1,56 @@
+import { Dispatch } from '@reduxjs/toolkit'
+import { Agent, Credential, Message } from 'daf-core'
+import { Callback, callbackify } from './util'
+import axios from 'axios'
+import { ActionSignW3cVc } from 'daf-w3c'
+import { addServiceToken } from '../reducers/authentication'
+
+export const serviceLoginFactory = (agent: Agent) => (serviceUrl: string, serviceDid: string, did: string, cb?: Callback<string>) => (dispatch: Dispatch) => callbackify(
+  async () => {
+    const makeLoginCredentialPayload = (challenge: string) => {
```

Comment:
I left them inside the `serviceAuthenticationFactory` because they depends on some variables that are received by the parent function. Do you think I should move them outside and send the needed params? I'm talking about `did` and `serviceDid`.

Reply:
We can curry them